optaplanner-docs/src/main/asciidoc/Chapter-Partitioned_search/Chapter-Partitioned_search.adoc

```diff
@@ -8,27 +8,214 @@
+        for (int i = 0; i < partCount; i++) {
+            CloudBalance partSolution = new CloudBalance(originalSolution.getId(),
+                    new ArrayList<>(originalSolution.getComputerList().size() / partCount + 1),
+                    new ArrayList<>(originalSolution.getProcessList().size() / partCount + 1));
+            partList.add(partSolution);
+        }
+        int partIndex = 0;
+        for (CloudComputer originalComputer : originalSolution.getComputerList()) {
+            CloudBalance part = partList.get(partIndex);
+            part.getComputerList().add(new CloudComputer(
+                    originalComputer.getId(),
+                    originalComputer.getCpuPower(), originalComputer.getMemory(),
+                    originalComputer.getNetworkBandwidth(), originalComputer.getCost()));
+            partIndex = (partIndex + 1) % partList.size();
+        }
+        partIndex = 0;
+        for (CloudProcess originalProcess : originalSolution.getProcessList()) {
+            CloudBalance part = partList.get(partIndex);
+            part.getProcessList().add(new CloudProcess(
+                    originalProcess.getId(),
+                    originalProcess.getRequiredCpuPower(), originalProcess.getRequiredMemory(),
+                    originalProcess.getRequiredNetworkBandwidth()));
+            if (originalProcess.getComputer() != null) {
+                // TODO switch the reference to the partition cloned computer
+                // TODO Then fail fast if the computer isn't in the same partition with this exception:
+                throw new IllegalStateException("The initialized process (" + originalProcess
+                        + ") has a computer (" + originalProcess.getComputer()
+                        + ") which belongs to the another partition.");
+            }
+            partIndex = (partIndex + 1) % partList.size();
+        }
+        return partList;
+    }
+
+}
+----
+
+
 [[runnablePartThreadLimit]]
 == Runnable Part Thread Limit
 
+When running a multi-threaded solver, such as Partitioned Search, CPU power can quickly become a scare resource,
+which can cause other processes or threads to hang or freeze.
+Luckily, Planner has a system to prevent CPU starving of
+other processes (such as an SSH connection in production or your IDE in development)
+or other threads (such as the servlet threads that handle REST requests).
+
+As explained in <<sizingHardwareAndSoftware, sizing hardware and software>>,
+each solver (including each child solver) does no IO during `solve()` and therefore saturates 1 CPU core completely.
+In Partitioned Search, every partition always has its own thread, called a part thread.
+It is impossible for 2 partitions to share a thread,
+because of <<asynchronousTermination,asynchronous termination>>: the second thread would never run.
+Every part thread will try to consume 1 CPU core entirely, so if there are more partitions than CPU cores,
+this would hang the system.
```

Comment:
Please spell out the numbers 1 and 2 where they occur in this paragraph. 

Also, "this *will* hang the system".